File: app/Modules/User/Routes/RoleRoutes.ts
import Route from '@ioc:Adonis/Core/Route'
import RolesController from 'App/Modules/User/Controllers/Http/Admin/RolesController'

Route.group(() => {
  Route.get('/roles', new RolesController().index).as('role.index')
  Route.get('/roles/:id', new RolesController().show).as('role.show')
})
  .prefix('admin')
  .middleware(['auth', 'acl:root,admin'])

File: src/app/modules/accounts/http/routes/role_routes.go
package routes

import (
	"github.com/gofiber/fiber/v2"
	"iecare-api/src/app/modules/accounts/http/controllers"
	"iecare-api/src/app/shared/http/middlewares"
)

func RoleRoutes(app *fiber.App, controller *controllers.RolesController) {
	api := app.Group("/roles")

	api.Use(middlewares.Acl([]string{"admin", "root"}))

	api.Get("/", controller.List).Name("roles.list")
	api.Get("/:roleId", controller.Get).Name("roles.get")
	api.Post("/", controller.Store).Name("roles.store")
	api.Put("/:roleId", controller.Edit).Name("roles.edit")
	api.Delete("/:roleId", controller.Delete).Name("roles.delete")
}
